ironfish/src/rpc/routes/chain/getTransaction.ts

```diff
@@ -72,36 +86,44 @@ router.register<typeof GetTransactionRequestSchema, GetTransactionResponse>(
     }
     const hashBuffer = BlockHashSerdeInstance.deserialize(request.data.blockHash)
 
-    const block = await node.chain.getBlock(hashBuffer)
-    if (!block) {
+    const blockHeader = await node.chain.getHeader(hashBuffer)
+    if (!blockHeader) {
       throw new ValidationError(`No block found`)
     }
 
     // Empty response used for case that transaction not found
     const rawTransaction: GetTransactionResponse = {
       fee: '0',
       expiration: 0,
+      index: 0,
       notesCount: 0,
       spendsCount: 0,
       signature: '',
       notesEncrypted: [],
       mints: [],
       burns: [],
     }
+    const transactions = await node.chain.getBlockTransactions(blockHeader)
 
-    block.transactions.map((transaction) => {
+    transactions.map(({ transaction, initialNoteIndex }) => {
```

Comment:
Can you revert all the above? All the data you need is already on the block header as `noteSize`

Reply:
Reason for this change is that i want to use an indeed same logic to get transaction initialNoteIndex as `wallet.connectBlock` does. I have get some errors caused by wrong calculation of transaction initialNoteIndex.

Reply:
What type of errors were you getting here @hairtail ?